crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/CallbackInterfaceImpl.ts

```diff
@@ -1,68 +1,72 @@
 {%- if self.include_once_check("CallbackInterfaceRuntime.ts") %}{%- include "CallbackInterfaceRuntime.ts" %}{%- endif %}
+{#
 {%- let trait_impl=format!("UniffiCallbackInterface{}", name) %}
 
 // Put the implementation in a struct so we don't pollute the top-level namespace
-fileprivate struct {{ trait_impl }} {
+class {{ trait_impl }} {
 
     // Create the VTable using a series of closures.
     // ts automatically converts these into C callback functions.
-    static var vtable: {{ vtable|ffi_type_name }} = {{ vtable|ffi_type_name }}(
+    static vtable: {{ vtable|ffi_type_name }} = new {{ vtable|ffi_type_name }}({
         {%- for (ffi_callback, meth) in vtable_methods %}
-        {{ meth.name()|fn_name }}: { (
+        {{ meth.name()|fn_name }}: (
             {%- for arg in ffi_callback.arguments() %}
             {{ arg.name()|var_name }}: {{ arg.type_().borrow()|ffi_type_name }}{% if !loop.last || ffi_callback.has_rust_call_status_arg() %},{% endif %}
             {%- endfor -%}
             {%- if ffi_callback.has_rust_call_status_arg() %}
             uniffiCallStatus: UnsafeMutablePointer<UniffiRustCallStatus>
             {%- endif %}
-        ) in
-            let makeCall = {
-                () {% if meth.is_async() %}async {% endif %}throws -> {% match meth.return_type() %}{% when Some(t) %}{{ t|type_name(ci) }}{% when None %}(){% endmatch %} in
-                guard let uniffiObj = try? {{ ffi_converter_name }}.handleMap.get(handle: uniffiHandle) else {
-                    throw UniffiInternalError.unexpectedStaleHandle
+        ) => {
+            const makeCall = {# space #}
+            {%- if meth.is_async() %}async {% else %} {% endif-%}
+            (): {% call ts::return_type(meth) %} => {
+                // Was converter.handleMap.get(uniffiHandle);
+                const uniffiObj = {{ ffi_converter_name }}.lift(uniffiHandle);
+                if (uniffiObj === undefined) {
+                    throw new UniffiInternalError.UnexpectedStaleHandle()
                 }
-                return {% if meth.throws() %}try {% endif %}{% if meth.is_async() %}await {% endif %}uniffiObj.{{ meth.name()|fn_name }}(
+                return {% if meth.is_async() %}await {% endif %}uniffiObj.{{ meth.name()|fn_name }}(
```

Comment:
so the await here is fine and totally acceptable, but not actually required if you are returning a Promise call directly.

I'm pretty sure Javascript/Typescript lets you return either Promise<Foo> or Foo in an async and will handle unwrapping the Promise.